### DISCUSSION TITLE
How to stop github actions workflow to trigger when the pull request is from dependabot-preview?

### DISCUSSION THREAD

**Discussion by riprasad** [community/community#26079] [Actions]
<p>Hi Team,</p>
<p>I have a workflow that gets triggered whenever somebody raises a pull request. However, I would not like the workflow to get triggered whenever their is an automated pull request from the <code>dependabot-preview</code> bot. How can I achieve that ?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/riprasad">@riprasad</a>,</p>
<p>Image pull request is created from compared branch ‘depbot’ to base branch. Typically workflow can be skipped with below methods:</p>
<ol>
<li>Use <a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">on.pull_request.branches/branches-ignore</a> to limit the base branch. code sample as below:</li>
</ol>
<pre><code class="lang-auto">on: 
  pull_request:
    branches-ignore: 
      - master                                 # pull request to master branch, event will NOT be triggered.
</code></pre>
<ol start="2">
<li>Add if expression in worklfow yaml, eg: skip the job if pull request comes from user dephot.  The limitation is the workflow will be listed in the actions tab but the job is skipped.</li>
</ol>
<pre><code class="lang-auto">on: [pull_request]
jobs:
  job1:
    if: github.actor!= 'depbot'   # ignore the pull request which comes from user depbot.
</code></pre>
<p>Note: you can change the if expression according to your needs.</p>
<ol start="3">
<li>If you don’t like whatever pull_request from depbot branch, you can comment out the pull_request event or delete it from yaml file, or even disable all github actions from repo setting- Actions tab.</li>
</ol>
<p>You can pick one according to your needs.</p>
<p>Thanks.</p>

**Comment by riprasad** [comment]
<p>Correction on the second point (correct actor name):-</p>
<pre><code class="lang-auto">on: [pull_request]
jobs:
  job1:
    if: github.actor!= 'dependabot-preview[bot]'   # ignore the pull request which comes from user depbot.
</code></pre>

**Comment by rseeton** [comment]
I'm not sure when/where this set, but our dependabot actor's name does not include the 'preview' portion.

```
on: [pull_request]
jobs:
  job1:
    if: github.actor!= 'dependabot[bot]'   # ignore the pull request which comes from user depbot.
```

**Comment by mconigliaro** [comment]
Does dependabot always create a branch named `dependabot/*`? If so, this might work better in a workflow with lots of jobs:

```
on: 
  pull_request:
    branches-ignore: 
      - dependabot/* 
```

**Reply by Mumeii** [reply]
@mconigliaro : Hi
Apparently, from what I see in the GH Action documentation, this branches-ignore feature only applies to the PR's targeted branche, not the source one.
